src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -434,13 +434,15 @@ type QsExpression with
             let openStartInSlicing =
                 function
                 | Some step when Some step |> validSlicing ->
-                    conditionalIntExpr (IsNegative step) (LengthMinusOne array) (SyntaxGenerator.IntLiteral 0L)
+                    let inferredStep = { step with ResolvedType = Int |> ResolvedType.New } // validSlicing ensures that the type must be an Int
```

Comment:
I would recommend moving the call to `inference.Resolve` out of `validSlicing` and instead have `integerExpr` return:

```fsharp
{ ex with ResolvedType = inference.Resolve ex.ResolvedType }
```

Then `validSlicing`, `openStartInSlicing`, and `openEndInSlicing` don't need to worry about resolving the type.

Reply:
I'm not sure what all that line affects, so let me know if it can be removed or not, please.